server/tests/api/video-blacklist-management.js

```diff
@@ -0,0 +1,235 @@
+const podsUtils = require('../utils/pods')
+const serversUtils = require('../utils/servers')
+const videosUtils = require('../utils/videos')
+const videoBlacklistsUtils = require('../utils/video-blacklists')
+
+describe('Test video blacklists management', function () {
+  let servers = []
+
+  before(function (done) {
+    this.timeout(120000)
+
+    series([
+      // Run servers
+      function (next) {
+        serversUtils.flushAndRunMultipleServers(2, function (serversRun) {
+          servers = serversRun
+          next()
+        })
+      },
+      // Get the access tokens
+      function (next) {
+        each(servers, function (server, callbackEach) {
+          loginUtils.loginAndGetAccessToken(server, function (err, accessToken) {
+            if (err) return callbackEach(err)
+
+            server.accessToken = accessToken
+            callbackEach()
+          })
+        }, next)
+      },
+      // Pod 1 makes friend with pod 2
+      function (next) {
+        const server = servers[0]
+        podsUtils.makeFriends(server.url, server.accessToken, next)
+      },
+      // Upload 2 videos on pod 2
+      function (next) {
+        videosUtils.uploadVideo(servers[1].url, servers[1].accessToken, { name: 'My 1st video', description: 'A video on pod 2' }, next)
+      },
+      function (next) {
+        videosUtils.uploadVideo(servers[1].url, servers[1].accessToken, { name: 'My 2nd video', description: 'A video on pod 2' }, next)
+      },
+      // Wait videos propagation
+      function (next) {
+        setTimeout(next, 22000)
+      },
+      // Blacklist the two videos on pod 1
+      function (next) {
+        videosUtils.getVideosList(servers[0].url, function (err, res) {
+          if (err) throw err
+
+          const videos = res.body.data
+
+          each(videos,
+               function (video, next) {
+		 videoBlacklistsUtils.addVideoToBlacklist(servers[0].url, servers[0].accessToken, video.id, next)
+	       },
+               next
+          )
+        })
+      }
+    ], done)
+  })
+
+  describe('When listing blacklisted videos', function () {
+    it('Should display all the blacklisted videos', function (done) {
+      videoBlacklistsUtils.getBlacklistedVideosList(servers[0].url, servers[0].accessToken, function (err, res) {
+        if (err) throw err
+
+        expect(res.body.total).to.equal(2)
+
+	const videos = res.body.data
+	expect(videos).to.be.an('array')
+	expect(videos.length).to.equal(2)
+
+	done()
+      })
+    })
+
+    it('Should get the correct sort when sorting by descending id', function (done) {
+      videoBlacklistsUtils.getSortedBlacklistedVideosList(servers[0].url, servers[0].accessToken, '-id', function (err, res) {
+        if (err) throw err
+
+        expect(res.body.total).to.equal(2)
+
+        const videos = res.body.data
+        expect(videos).to.be.an('array')
+        expect(videos.length).to.equal(2)
+
+        const result = res.body.data.slice(0).sort((a, b) => {
```

Comment:
Try to use https://lodash.com/docs/4.17.4#sortBy

And why do you use slice?

Reply:
Because in the test I want to check if the two arrays have the same the content. I don't want to check if they have the same reference so I'm not using `expect(videos).equals(result)`.

To be sure the `videos` array have the same content as `result`, while being a different object, I make a sorted copy of the `res.body.data` object (which is the result I expect in the test). And `slice(0)` is the fastest way to make a copy of an array in Javascript according to many benchmarks.

Reply:
👍

Reply:
I will use the [orderBy](https://lodash.com/docs#orderBy) method because it allows to specify sort orders. Also, this method creates a new array so there's no need to use `slice(0)` anymore